services/local-notifications/core.go

```diff
@@ -474,16 +476,33 @@ func (s *Service) IsStarted() bool {
 	return s.started
 }
 
-func SendMessageNotifications(mnb []protocol.MessageNotificationBody) {
+func showNotification(publicKey ecdsa.PublicKey, n protocol.MessageNotificationBody) bool {
```

Comment:
Personally I would move this logic closer to `messenger.go`,
as it's very contextual (it will deal with `chats`, `contacts`, `communities` etc).
But up to you, either ways it's ok.
Could you please add a test for this functionality?

Reply:
I have some code in https://github.com/status-im/status-go/pull/2142 that will give this code a better home.

We could either add it below `message.New = true` :

https://github.com/status-im/status-go/blob/f9d08bf9c272022f9e4a06082aa8acf9c17beb85/protocol/messenger.go#L3259-L3268

Or inside :

https://github.com/status-im/status-go/blob/f9d08bf9c272022f9e4a06082aa8acf9c17beb85/protocol/messenger.go#L2756-L2779

Reply:
code is moved